salt/modules/debbuild.py

```diff
@@ -416,20 +442,66 @@ def make_repo(repodir, keyid=None, env=None):
     with salt.utils.fopen(repoconfdist, 'w') as fow:
         fow.write('{0}'.format(repocfg_dists))
 
+    local_fingerprint = None
+    local_keyid = None
+
     if keyid is not None:
         with salt.utils.fopen(repoconfdist, 'a') as fow:
             fow.write('SignWith: {0}\n'.format(keyid))
 
+        # gpg keys should have been loaded as part of setup
+        # retrieve specified key, obtain fingerprint and preset passphrase
+        local_keys = __salt__['gpg.list_keys'](user=runas, gnupghome=gnupghome)
+        for gpg_key in local_keys:
+            if keyid == gpg_key['keyid'][8:]:
+                local_fingerprint = gpg_key['fingerprint']
+                local_keyid = gpg_key['keyid']
+                break
+
+        if local_keyid is None:
+            raise SaltInvocationError(
+                '\'keyid\' was not found in gpg keyring'
+            )
+
+        # preset passphase and interaction with gpg-agent
+        gpg_info_file = '{0}/gpg-agent-info-salt'.format(gnupghome)
+        with salt.utils.fopen(gpg_info_file, 'r') as fow:
```

Comment:
I don't see where this is getting written out above. Is it expected that this file already exists before this function is run? If not, it seems like this would raise an `IOError`.